node/core/prospective-parachains/src/lib.rs

```diff
@@ -872,11 +880,24 @@ async fn fetch_ancestry<Context>(
 				);
 
 				// Return, however far we got.
-				return Ok(block_info)
-			},
-			Some(info) => {
-				block_info.push(info);
+				break
 			},
+			Some(info) => info,
+		};
+
+		// The relay chain cannot accept blocks backed from previous sessions, with
+		// potentially previous validators. This is a technical limitation we need to
+		// respect here.
```

Comment:
Comment explaining this logic here.

This change is probably more significant to push to collators, as they will see their unincluded segment as being full at session boundaries without it.